From d137854b2bda5843aff2c815917a6a3ee5c9fe92 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 8 Feb 2016 19:51:27 +0200
Subject: [PATCH 0835/1240] fix: comphy: sata: cp110: get base address of SATA
 unit from FDT in COMPHY

- No need to define SATA base address - the base address already defined in
  device tree

Change-Id: I775f4dc1a4fd62f60fa00f109de5388feddef303
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27362
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/include/asm/arch-armada8k/regs-base.h | 1 -
 drivers/phy/comphy_cp110.c                     | 8 ++++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/include/asm/arch-armada8k/regs-base.h b/arch/arm/include/asm/arch-armada8k/regs-base.h
index 49c2e21..3e72d3b 100644
--- a/arch/arm/include/asm/arch-armada8k/regs-base.h
+++ b/arch/arm/include/asm/arch-armada8k/regs-base.h
@@ -46,6 +46,5 @@
 
 /* Base registers for CP */
 #define MVEBU_AMB_IP_BASE		(MVEBU_CP0_REGS_BASE + 0x13ff00)
-#define MVEBU_SATA3_GENERAL_BASE	(MVEBU_CP0_REGS_BASE + 0x540000)
 
 #endif	/* _REGS_BASE_H_ */
diff --git a/drivers/phy/comphy_cp110.c b/drivers/phy/comphy_cp110.c
index 0164d71..48810ca 100644
--- a/drivers/phy/comphy_cp110.c
+++ b/drivers/phy/comphy_cp110.c
@@ -339,8 +339,12 @@ static int comphy_sata_power_up(u32 lane, void __iomem *hpipe_base, void __iomem
 	void __iomem *sata_base;
 
 	debug_enter();
-	sata_base = (void __iomem *)MVEBU_SATA3_GENERAL_BASE;
-	debug("SATA3 General address base %p\n", sata_base);
+	sata_base = fdt_get_reg_offs_by_compat(COMPAT_MVEBU_SATA);
+	if (sata_base == 0) {
+		debug("SATA address not found in FDT\n");
+		return 0;
+	}
+	debug("SATA address found in FDT %p\n", sata_base);
 
 	debug("stage: MAC configuration - power down comphy\n");
 	/* MAC configuration powe down comphy
-- 
1.9.1

